dtm0/service.c

```diff
@@ -306,6 +303,21 @@ M0_INTERNAL bool m0_dtm0_is_a_persistent_dtm(struct m0_reqh_service *service)
 		to_dtm(service)->dos_origin == DTM0_ON_PERSISTENT;
 }
 
+M0_INTERNAL int m0_dtm0_service_find(struct m0_reqh *reqh,
```

Comment:
thanks, returning a pointer and adding `const` qualifier makes sense here.

to_dtm() will not take care about nullptr. It will panic because bob_of will panic. It has to be a bit more sophisticated than you have described.